docsrc/examples/VI as Sampler Inits.ipynb

```diff
@@ -0,0 +1,173 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The ADVI algorithm provides estimates of all model parameters as well as the step size scaling factor `eta`.\n",
+    "\n",
+    "The `variational` method returns a `CmdStanVB` object, with methods `eta` and `stan_variables`, which\n",
+    "return the step size scaling factor and estimates of all model parameters as a Python dictionary respectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(vb_fit.eta, vb_fit.stan_variables())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Posteriordb provides reference posteriors for all models.  For the logearn_height model, conditioned on the dataset `earnings.json`, the posterior variables are:\n",
+    "\n",
+    "- beta[1]:  5.782\n",
+    "- beta[2]:  0.059\n",
+    "- sigma: 0.894\n",
+    "\n",
+    "By default, the sampler algorithm randomly initializes all model parameters in the range uniform[-2, 2].  The ADVI estimates will provide a better starting point, especially w/r/t to parameter `beta[1]`, than the defaults.\n",
+    "In addition, we can use the step size scaling factor to scale the initial step size, which allows us to skip the first phase of warmup (default 75 iterations)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vb_vars = vb_fit.stan_variables()\n",
+    "vb_stepsize = 1.0 / vb_fit.eta\n",
+    "mcmc_vb_inits_fit = model.sample(\n",
+    "    data=data_file, inits=vb_vars, step_size=vb_stepsize,\n",
+    "    adapt_init_phase=0, seed=123\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mcmc_vb_inits_fit.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The sampler results match the reference posterior, (taking into account MCSE).\n",
+    "\n",
+    "- beta[1]:  5.782\n",
+    "- beta[2]:  0.059\n",
+    "- sigma: 0.894\n",
+    "\n",
+    "To see how this is useful, we run the sampler with default initializations, step size, and warmup, (same random seed)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mcmc_random_inits_fit = model.sample(data=data_file, seed=123)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mcmc_random_inits_fit.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "On my machine, using the variational estimates to skip warmup phase I shows improved N_Eff/s (number of effective sampler per second) values for all parameters.\n",
```

Comment:
This may be my overly academic bias, but I prefer to keep 'I' 'my' etc out of the doc. Happy with the royal we used elsewhere in this file. If we're pinning seeds, though, should we expect better N_Eff/s on all machines?

Reply:
correct on all counts - done